PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -188,7 +188,7 @@ struct HfTrackIndexSkimCreatorTagSelCollisions {
   PROCESS_SWITCH(HfTrackIndexSkimCreatorTagSelCollisions, processTrigSel, "Use trigger selection", false);
 
   /// Event selection without trigger selection
-  void processNoTrigSel(aod::Collision const& collision, aod::BCsWithTimestamps const& bcs)
+  void processNoTrigSel(soa::Join<aod::Collisions, aod::EvSels>::iterator const& collision, aod::BCsWithTimestamps const& bcs)
```

Comment:
is it necessary? In principle everything is templetized properly, and even w/o the `Join` the compilation should not complain. Even, this is the function w/o "trigger sleections" meant as "event selections" (see the `false` template parameter, which consequently invokes `hfEvSel.getHfCollisionRejectionMask<applyTrigSel, ...>()` that does not require the `Join` with `aod::EvSels`). I'd leave this piece untouched.

Reply:
Hi @mfaggin , the `processNoTrigSel` function calls `hfEvSel.getHfCollisionRejectionMask` which with this PR needs the occupancy information stored in `aod::EvSels`. If I do not join the tables, it breaks at compilation time. If I understood correctly, the joining should not have any effect since the  `applyTrigSel` is still false and avoids the event selections. Of course if you have a better solution I will implement it rightaway!

Reply:
Ciao @stefanopolitano are you sure that the compilation fails? Have you explicitly seen it? This would be surprising to me, since the function is templetized (see `typename Coll` here https://github.com/AliceO2Group/O2Physics/blob/2363ac8e965e9b6f15a218926b31c32b20bfe32b/PWGHF/Utils/utilsEvSelHf.h#L162-L163). This would surprise me, honestly.

What I see that it can be causing some troubles is the fact that with this PR you add the `TH2` to monitor the selected events vs. centrality and occupancy, and the occupancy info is indeed in the `EvSel` table. However, this issue affects only the function `fillHistograms`, and in my opinion you can overcome it easily as for the `centrality`, i.e. defining a default argument `float occupancy = -1` that you agnostically use to fill the `TH2` (in other words: the decision to use `collision.ft0cOccupancyInTimeRange()` or `collision.trackOccupancyInTimeRange()` is moved when calling the function itself).

Reply:
Hi, I see your point. Ok then, I will modify the code to have a default occupancy value. I am adding the occupancy info in the creators and in the flow task to proceed with the occupancy study. Thanks!